api/src/main/resources/sql/procedures/create_functions.sql

```diff
@@ -907,3 +907,75 @@ RETURN ret;
 END
 
 #
+
+-- This function accepts encounter_id, mapping source, mapping code, offset value
+-- It will find a single, best observation that matches this, and return the value_datetime
```

Comment:
Maybe more description about where the offset fits in - what use case is this solving?  when would you choose to use this function?